src/main/frontend/src/index.js

```diff
@@ -58,15 +59,23 @@ const store = configureStore({
 // Once the store is configured, load the courses from the backend.
 store.dispatch(fetchCourses(1));
 
-root.render(
-  <React.StrictMode>
-    <Provider store={store}>
-      <ThemeProvider breakpoints={['xs', 'xxs','sm']}>
-        <App />
-      </ThemeProvider>
-    </Provider>
-  </React.StrictMode>
-);
+let stateAndNonceCheck = new LtiStorage().validateStateAndNonce(ltiLaunchData.state, ltiLaunchData.nonce, ltiLaunchData.iss, window);
+stateAndNonceCheck.then(function(result) {
+    if (result === true){
+      root.render(
+        <React.StrictMode>
+          <Provider store={store}>
+            <ThemeProvider breakpoints={['xs', 'xxs','sm']}>
+              <App />
+            </ThemeProvider>
+          </Provider>
+        </React.StrictMode>
+      );
+    }
```

Comment:
Could we add a comment here that if `result === false` this is where we'll want to redirect to the error page?